### PR TITLE
[Enhancement] Use object instead of pointer for HyperLogLog::_hash_set

### PR DESCRIPTION
**Pull request by ZiheLiu**
## What type of PR is this：
- [ ] bug
- [ ] feature
- [x] enhancement
- [ ] refactor
- [ ] others

## Problem Summary(Required) ：

#6885 uses pointer instead of object for `HyperLogLog::_hash_set` to reduce memory usage of HLL and not include `phmap.h` in `hll.h`.

However, as for reading HLL column from disk, 
- using pointer will allocate memory 4096 times for  HLL hash_set of 4096 rows in a chunk,
- while using object only need allocate 1 times for a chunk by `_pool.reserve`.

```C++
template <typename T>
bool ObjectColumn<T>::append_strings(const Buffer<starrocks::Slice>& strs) {
    _pool.reserve(_pool.size() + strs.size());
    for (const Slice& s : strs) {
        _pool.emplace_back(s);
    }
```

And using object doesn't cost much more than using pointer.
- the empty `flat_hash_set<uint64_t>` is not very large, only with 32 bytes, 
- while using pointer costs 8 bytes.
```C++
    ctrl_t* ctrl_ = EmptyGroup(); // [(capacity + 1) * ctrl_t]
    slot_type* slots_ = nullptr;  // [capacity * slot_type]
    size_t size_ = 0;             // number of full slots
    size_t capacity_ = 0;         // total number of slots
```

Therefore, this PR uses object instead of pointer again.

### Environment
- 3 BE (each 16 vCPUs and 64GB RAM)
- ssb_100g
- Query is as follows, where `v2` is  a `HLL` column, table contains 6e8 rows.
```sql
select hll_union_agg(v2) / 2 as a from lineorder_flat_bitmap_hll
```

### Result
- 2.1: 1.5s
- main-Before PR: 3s
- main-After PR: 1.5s

### CODE REVIEW

File: be/src/types/hll.h [hunk 1]
```diff
@@ -135,7 +135,7 @@ class HyperLogLog {
     HllDataType _type = HLL_DATA_EMPTY;
 
     using ElementSet = phmap::flat_hash_set<uint64_t>;
-    std::unique_ptr<ElementSet> _hash_set;
+    ElementSet _hash_set;
```
Comment (imay) [inline be/src/types/hll.h:138]: should add comment for this optimization

Comment (ZiheLiu) [reply be/src/types/hll.h:141]: Thanks, fixed.

**Review by imay — APPROVED** [review]

**Comment by ZiheLiu** [comment]
run starrocks_admit_test

**Review by satanson — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Pull request merged**